File: exercises/00 Writing a first unit test/00 Writing a first unit test/StrangeNumber.cs
namespace _00_Writing_a_first_unit_test
{
    public class StrangeNumber
    {
        private double value = 0.0;

        public double GetDouble()
        {
            return value;
        }

        public void SetDouble(double value)
        {
            this.value = value;
        }

        public void Add(double i)
        {
            value += i;
        }

        public void Subtract(double i)
        {
            value -= i;
        }

        public void Divide(double i)
        {
            value /= i;
        }

        public void Multiply(double i)
        {
            value *= i;
        }

    }
}

File: exercises/00 Writing a first unit test/C++/00 Writing a first unit test/StrangeNumber.cpp
#include "StrangeNumber.h"

StrangeNumber::StrangeNumber() :
	_value(0.0)
{
}

StrangeNumber::~StrangeNumber()
{
}

double StrangeNumber::getDouble() const
{
	return _value;
}

void StrangeNumber::setDouble(double value)
{
	_value = value;
}

void StrangeNumber::add(double value)
{
	_value += value;
}

void StrangeNumber::substract(double value)
{
	_value -= value;
}

void StrangeNumber::divide(double value)
{
	_value /= value;
}

void StrangeNumber::multiply(double value)
{
	_value *= value;
}
